.github/workflows/rust.yml

```diff
@@ -1,72 +1,57 @@
 name: rust
 
 on:
-  push:
-    branches: [main, staging, trying]
-  pull_request:
-    branches: [main]
+    push:
+        branches: [main, staging, trying]
+    pull_request:
+        branches: [main]
 
 env:
-  CARGO_INCREMENTAL: 0
-  CARGO_TERM_COLOR: always
-  CI: 1
-  RUSTFLAGS: "-W rust-2021-compatibility -D warnings"
-  RUST_BACKTRACE: short
+    CARGO_INCREMENTAL: 0
+    CARGO_TERM_COLOR: always
+    CI: 1
+    RUSTFLAGS: "-W rust-2021-compatibility -D warnings"
+    RUST_BACKTRACE: short
 
 jobs:
-  check-format:
-    # Skip draft release PRs
-    if: ${{ github.actor_id != '166155226' || github.event_name != 'pull_request' || github.event.pull_request.draft == false }}
-
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: dtolnay/rust-toolchain@1.76.0
-        with:
-          components: rustfmt
-      - name: Check formatting
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: -- --check
-
-  build:
-    # Skip draft release PRs
-    if: ${{ github.actor_id != '166155226' || github.event_name != 'pull_request' || github.event.pull_request.draft == false }}
-
-    runs-on: ubuntu-latest
-
-    steps:
-      - uses: actions/checkout@v2
-      - uses: dtolnay/rust-toolchain@1.76.0
-      - uses: Swatinem/rust-cache@v2
-      - name: Build
-        shell: bash
-        run: |
-          cargo build --all-features
-
-      - name: Build tests
-        shell: bash
-        run: |
-          cargo test --all-features --no-run
-
-      - name: Run tests
-        shell: bash
-        run: |
-          cargo test --all-features
-
-      - name: Build examples
-        shell: bash
-        run: |
-          cargo build --all --examples
-
-      - name: Build example tests
-        shell: bash
-        run: |
-          cargo test --all --examples --no-run
-          cargo test --all --examples
-
-      - name: Run example tests
-        shell: bash
-        run: |
-          cargo test --all --examples
+    check:
+        # Skip draft release PRs
+        if: github.actor_id != '166155226' || github.event_name != 'pull_request' || github.event.pull_request.draft == false
+
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0
+            - uses: dtolnay/rust-toolchain@1.76.0
+              with:
+                  components: clippy, rustfmt
+            - name: Check format
+              run: cargo fmt --all --check
+            - name: Check lint
+              run: cargo clippy --all --all-features
```

Comment:
This job was specifically for checking formats.  I'm not against running clippy on CI, but it should be part of the build step so it can share build artifacts with all the other builds.

So can you move this into the build section please